upup/pkg/fi/cloudup/gce/gce_cloud.go

```diff
@@ -29,24 +30,41 @@ import (
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns"
 )
 
-type GCECloud struct {
-	Compute *compute.Service
-	Storage *storage.Service
+type GCECloud interface {
```

Comment:
Ci is broke cause your awesome tests got broken, may need some help to wire in the new interface